src/datasets/arrow_dataset.py

```diff
@@ -424,55 +430,31 @@ def to_tf_dataset(
             if col not in output_signature:
                 raise ValueError(f"Label column {col} not found in dataset!")
 
-        def np_get_batch(indices):
-            # Optimization - if we're loading a sequential batch, do it with slicing instead of a list of indices
-            if np.all(np.diff(indices) == 1):
-                batch = dataset[indices[0] : indices[-1] + 1]
-            else:
-                batch = dataset[indices]
-
-            if cols_to_retain is not None:
-                batch = {
-                    key: value
-                    for key, value in batch.items()
-                    if key in cols_to_retain or key in ("label", "label_ids", "labels")
-                }
-            elif cols_to_retain is not None:
-                batch = {key: value for key, value in batch.items() if key in cols_to_retain}
-
-            actual_size = len(list(batch.values())[0])  # Get the length of one of the arrays, assume all same
-            # Our collators expect a list of dicts, not a dict of lists/arrays, so we invert
-            batch = [{key: value[i] for key, value in batch.items()} for i in range(actual_size)]
-            batch = collate_fn(batch, **collate_fn_args)
-            out_batch = []
-            for col, cast_dtype in columns_to_np_types.items():
-                # In case the collate_fn returns something strange
-                array = np.array(batch[col])
-                array = array.astype(cast_dtype)
-                out_batch.append(array)
-            return out_batch
-
-        @tf.function(input_signature=[tf.TensorSpec(None, tf.int64)])
-        def fetch_function(indices):
-            output = tf.numpy_function(
-                np_get_batch,
-                inp=[indices],
-                # This works because dictionaries always output in the same order
-                Tout=[tf.dtypes.as_dtype(dtype) for dtype in columns_to_np_types.values()],
+        if num_workers <= 0:
```

Comment:
Bit picky, but I would do something like (unless a negative number indicates something special?)

```
if num_workers == 0:
    ...
elif num_workers >= 1:
    ...
else:
    raise ValueError(f"Invalid number of workers {num_workers}. Must be >= 0")
```

Reply:
Haha, no, this time negatives do not have special meaning! Your approach is better, I'll make the change